x-pack/plugins/secops/public/components/page/hosts/authentications_table/index.tsx

```diff
@@ -93,10 +97,16 @@ const AuthenticationTableComponent = pure<AuthenticationTableProps>(
   )
 );
 
-const mapStateToProps = (state: State) => authenticationsSelector(state);
+const makeMapStateToProps = () => {
+  const getAuthenticationsSelector = hostsSelectors.authenticationsSelector();
+  const mapStateToProps = (state: State, { type }: OwnProps) => {
+    return getAuthenticationsSelector(state, type);
+  };
+  return mapStateToProps;
+};
```

Comment:
Is this a trick with redux where it's checking the arity/number of arguments in the function and so you're wrapping this with no arguments for that?

Otherwise if redux just calls this normally to get a function and you really want the authenticationsSelector to be a function wouldn't all of this be just:

```ts
const mapStateToProps = (state: State, { type }: OwnProps) =>
  hostsSelectors.authenticationsSelector()(state, type);
```

Where `authenticationsSelector()` has an extra function call?

Reply:
advised by the library -> https://github.com/reduxjs/reselect#componentsappjs

Reply:
Per conversations, this is from:
https://github.com/reduxjs/reselect#sharing-selectors-with-props-across-multiple-component-instances

And there is also this viable way which is `re-reselector`. Either way does the job needed.

Reply:
Optional. If the closure is retained you could keep this a bit smaller with this which should do the same thing:

```ts
const makeMapStateToProps = () => {
  const getAuthenticationsSelector = hostsSelectors.authenticationsSelector();
  return (state: State, { type }: OwnProps) => getAuthenticationsSelector(state, type);
};
```

Reply:
I will keep it the way it is because when I wrote it. It did help me to understand the code with mapStateToProps and I do think that it will help other people later on the way it is